src/System.CommandLine/Command.cs

```diff
@@ -251,6 +254,18 @@ void AddCompletionsFor(Symbol identifier, AliasSet? aliases)
             }
         }
 
+        /// <summary>
+        /// Sets a command's handler based on an <see cref="Action{InvocationContext}"/>.
+        /// </summary>
+        public void SetHandler(Action<InvocationContext> handle)
+            => Handler = new AnonymousCommandHandler(handle);
+
+        /// <summary>
+        /// Sets a command's handler based on a <see cref="Func{Task,InvocationContext}"/>.
+        /// </summary>
+        public void SetHandler(Func<InvocationContext, CancellationToken, Task> handle)
+            => Handler = new AnonymousCommandHandler(handle);
```

Comment:
> If that feature is going to be preserved, it should be documented here.

Our goal is to remove `InvocationContext` and have an int-returning delegate that accepts a `ParseResult` and `CancellationToken` (in case of the async handler)

> Besides, the reference to Func{InvocationContext, Task} is not correct as there is a CancellationToken as well.

Good point, I've updated the comment.

Reply:
> The way AnonymousCommandHandler checks whether the delegate returned a Task<int> rather than just a Task is rather surprising.

This was in place to support some of the reflection-based approaches that had been in approach in beta1 and should no longer be needed.

Reply:
> Does https://github.com/dotnet/command-line-api/issues/2071 cover the removal of the Task<int> type check or should I file a separate issue?

I've sent a PR that makes it mandatory for all the handlers to return an integer:

https://github.com/dotnet/command-line-api/pull/2092